### PR TITLE
Issue/349 fill name nickname and email

### LINKED ISSUE #349: Fill `name`, `nickname` and `email` occurrences in `user-svc` with data from the actual user entity
### Describe the chore

Example https://github.com/helpwave/services/blob/main/services/user-svc/internal/organization/organization.go#L203

### PR DESCRIPTION
**Pull request by florianjs**
## Which issues does this pull request close?
closes #349

### CODE REVIEW

File: services/user-svc/internal/repositories/user_repository.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+func UserRepo(logCtx context.Context) *UserRepository {
+	return &UserRepository{
+		db:  hwgorm.GetDB(logCtx),
+		ctx: logCtx,
+	}
+}
+
+func (r *UserRepository) GetUserById(id uuid.UUID) (*models.User, error) {
+
```
Comment (use-to) [inline services/user-svc/internal/repositories/user_repository.go:25]:
```suggestion
```

File: services/user-svc/internal/models/user_models.go [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package models
+
+import "github.com/google/uuid"
+
+type BaseUser struct {
+	Email    string `gorm:"column:email"`
+	Nickname string `gorm:"column:nickname"`
+	Name     string `gorm:"column:name"`
```
Comment (use-to) [inline services/user-svc/internal/models/user_models.go:8]: Just another spontanious idea: Add `AvatarUrl` here.
On create generate a sha-hash of the Email.
Insert `https://source.boringavatars.com/marble/128/{hash}` into the database (this is always the same picture).

I feel like this is really easy to implement and will save us a lot of time later on.

**Review by use-to — CHANGES_REQUESTED** [review]
Everything is perfect. I would love to have the avatar url implementation on top.

File: services/user-svc/internal/user/user.go [hunk 3]
```diff
@@ -96,13 +53,14 @@ func HandleUserRegisteredEvent(ctx context.Context, evt *daprcmn.TopicEvent) (re
 		return true, err
 	}
 
-	db := hwgorm.GetDB(ctx)
+	var user *models.User
+	if existingUser, _ := userRepository.GetUserById(userID); existingUser == nil {
+		avatarUrl := fmt.Sprintf("%s%d", BoringAvatarURL, userID)
```
Comment (MaxSchaefer) [inline services/user-svc/internal/user/user.go:58]: Please hash the ID and move this into a separate repository function like generate `generateDefaultAvatarUrl(...)`

File: services/user-svc/main.go [hunk 4]
```diff
@@ -24,8 +24,8 @@ func main() {
 
 		daprClient := common.MustNewDaprGRPCClient()
 
-		common.MustAddTopicEventHandler(server, user.UserRegisteredEventSubscription, user.HandleUserRegisteredEvent)
-		common.MustAddTopicEventHandler(server, user.UserUpdatedEventSubscription, user.HandleUserUpdatedEvent)
+		common.MustAddTopicEventHandler(server, user.RegisteredEventSubscription, user.HandleUserRegisteredEvent)
+		common.MustAddTopicEventHandler(server, user.UpdatedEventSubscription, user.HandleUserUpdatedEvent)
```
Comment (MaxSchaefer) [inline services/user-svc/main.go:28]:
```suggestion
		common.MustAddTopicEventHandler(server, user.RegisteredEventSubscription, user.HandleUserRegisteredEvent)
		common.MustAddTopicEventHandler(server, user.UpdatedEventSubscription, user.HandleUserUpdatedEvent)
```
Why was that changed? Not in scope of this pull request.

File: services/user-svc/migrations/000007_user_avatar_url.up.sql [hunk 5]
```diff
@@ -0,0 +1,3 @@
+ALTER TABLE users
+	ADD COLUMN IF NOT EXISTS
+	    avatar_url text;
```
Comment (MaxSchaefer) [inline services/user-svc/migrations/000007_user_avatar_url.up.sql:3]: You could default here to a to static URL of boring avatars.

File: services/user-svc/migrations/000007_user_avatar_url.down.sql [hunk 6]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE users
+	DROP COLUMN IF EXISTS avatar_url ;
```
Comment (MaxSchaefer) [inline services/user-svc/migrations/000007_user_avatar_url.down.sql:2]:
```suggestion
	DROP COLUMN IF EXISTS avatar_url;
```

**Review by MaxSchaefer — CHANGES_REQUESTED** [review]

File: services/user-svc/main.go (hunk 4 above)
Comment (florianjs) [reply services/user-svc/main.go:28]: Yes is not part of this pull request however it should be avoided that functions with the package name staten

Comment (florianjs) [reply services/user-svc/main.go:28]: Yes is not part of this pull request however it should be avoided that functions start with the package name

**Review by MaxSchaefer — APPROVED** [review]

**Comment by MaxSchaefer** [comment]
@use-to

**Comment by use-to** [comment]
> @use-to please check your request for changes.

Sure. I have no more points. But you initiated a thread - still unresolved. Is the renaming okay?

**Comment by florianjs** [comment]
> > @use-to please check your request for changes.
> 
> Sure. I have no more points. But you initiated a thread - still unresolved. Is the renaming okay?

Yes i think so. Because he approved

**Review by use-to — APPROVED** [review]
Great job!
